test/test_auth.py

```diff
@@ -101,6 +101,44 @@ def test_default_provider(self):
             self.assertTrue(EXAMPLE_SECRET_ACCESS_KEY == credentials.secret_access_key)
             self.assertTrue(credentials.session_token is None)
 
+    def _prepare_profile_file(self, profile_name, credential_file_name):
+        # prepare a credential profile file for the credential provider
+        file = open(credential_file_name, "w")
+        file.write("[default]\n" +
```

Comment:
Seems simpler to just commit a real actual file to `test/resources/example_profile`?

If you need to write the file from code, use something from [tempfile](https://docs.python.org/3/library/tempfile.html) to create a temporary file in the OS's appropriate tmp directory